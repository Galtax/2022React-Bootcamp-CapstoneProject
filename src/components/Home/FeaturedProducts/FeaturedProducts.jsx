import React from 'react';
import { useGetSearch } from '../../../utils/hooks/useFetch';
import { ProductCard } from '../../ProductCard/ProductCard';
import { Spiner } from '../../ProductList/ProductList.styles';
import { ProductGrid } from './FeaturedProducts.styled';

function FeaturedProducts() {

  const documentType = '"product"';
 
  const { data, isLoading } = useGetSearch({documentType, pageSize: 16});
  
  const allItems = data?.results?.map((item) => {
    return <ProductCard item={item} key={item.id}/>;
  })
  return (
    <>
      <ProductGrid>
        {isLoading
          ? <Spiner />
          : allItems}
      </ProductGrid>
      
    </>
  )
}

export default FeaturedProducts;
