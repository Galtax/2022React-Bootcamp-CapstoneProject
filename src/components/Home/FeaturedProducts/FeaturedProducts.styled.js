import styled from "styled-components";

export const ProductGrid = styled.div`
  background: none;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
`;
