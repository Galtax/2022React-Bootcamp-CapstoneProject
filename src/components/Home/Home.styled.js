import styled from 'styled-components';

export const HomeMain = styled.main`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 13rem 5rem;
  max-width: 100%;
  min-height: 95vh;

  @media (max-width: 576px) {
    margin: 7rem 2rem;
  }

  @media (min-width: 577px) and (max-width: 1024px) {
    margin: 4rem 5rem;
  }
`;