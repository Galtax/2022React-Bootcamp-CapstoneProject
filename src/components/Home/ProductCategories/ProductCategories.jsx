import React from 'react';
import { ROUTES } from '../../../utils/constants';
import { useGetSearch } from '../../../utils/hooks/useFetch';
import { Spiner } from '../../ProductList/ProductList.styles';
import { ProductGrid, ProductItem } from './Productcategories.styled';

function ProductCategories() {

const documentType = '"category"';
const { data, isLoading } = useGetSearch({documentType, pageSize: 30});

const products = data?.results?.map((item, i) => {
  const name = item.data.name.replaceAll(' & ', '--');
  return(
    <ProductItem 
      to={
        `${ROUTES.productList}?category=${name}`} key={`product-${i}`}>
      {item.data.name}
    </ProductItem>
  )
})
  
  return(
    <ProductGrid>
      {isLoading
      ? <Spiner />
      : products}
    </ProductGrid>
  );
}

export default ProductCategories;
