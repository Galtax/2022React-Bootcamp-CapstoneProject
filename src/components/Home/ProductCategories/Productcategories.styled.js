import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { bluePrince, ligthPrinceBlue } from '../../../utils/constant.styles';

export const ProductGrid = styled.div`
  display: flex;
  justify-content: space-around;
  background: rgba(250,250,250,.4);
  margin: 2.5rem 0;

  @media screen and (max-width: 600px){
    margin: 1.5rem 0;
    flex-flow: row wrap;
    max-height: fit-content;
  }
`;

export const ProductItem = styled(Link)`
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 2rem auto;
  width: 100%;
  height: 2rem;
  font-size: 1.3rem;
  font-weight: 400;
  color: ${bluePrince};
  text-decoration: none;

  &:hover {
    color: ${ligthPrinceBlue};
    text-decoration: underline;
  }

  &:not(:first-child){
    border-left: 1px solid ${bluePrince};
  }

  @media screen and (max-width: 1100px){
    font-size: 1.1rem;
  }

  @media screen and (max-width: 900px){
    &:not(:first-child){
      border-left: none; 
      margin-right: .2rem;
    } 
  }
  @media screen and (max-width: 600px){
   width: fit-content;
   margin: 0 1rem;
   height: 3.5rem;
  }
`;
