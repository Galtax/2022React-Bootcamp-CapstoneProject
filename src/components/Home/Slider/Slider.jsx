import React, { useRef } from 'react';
import { useFeaturedBanners } from '../../../utils/hooks/useFeaturedBanners';
import { ReactComponent as LeftArrowIcon } from '../../../images/iconmonstr-angel-left-thin.svg';
import { ReactComponent as RightArrowIcon } from '../../../images/iconmonstr-angel-right-thin.svg';
import { MainBox, SlideshowBox, Slide, Controls, Button } from './Slider.styled';
import { Spiner } from '../../ProductList/ProductList.styles';

function Slider() {
  const slideshow = useRef(null);

  const nextPic = () => {
    if (slideshow.current.children.length > 0) {
      const firstElement = slideshow.current.children[0];

      slideshow.current.style.transition = '700ms ease-out all';
      slideshow.current.style.transform = 'translateX(-100%)';

      const transitionTime = () => {
        slideshow.current.style.transition = 'none';
        slideshow.current.style.transform = 'translateX(0)';
        slideshow.current.appendChild(firstElement);
        slideshow.current.removeEventListener('transitionend', transitionTime);
      };
      slideshow.current.addEventListener('transitionend', transitionTime);
    }
  };

  const lastPic = () => {
    if (slideshow.current.children.length > 0) {
      const lastPosition = slideshow.current.children.length - 1;
      const lastElement = slideshow.current.children[lastPosition];

      slideshow.current.insertBefore(lastElement, slideshow.current.firstChild);
      slideshow.current.style.transition = 'none';
      slideshow.current.style.transform = 'translateX(-100%)';

      setTimeout(() => {
        slideshow.current.style.transition = '700ms ease-out all';
        slideshow.current.style.transform = 'translateX(0)';
      }, 30);
    }
  };

  const { data, isLoading } = useFeaturedBanners();
 

  const banners = data?.results?.map((item, i) => (
    <Slide key={`slide-${i}`}>
      <img src={item.data.main_image.url} alt="" />
    </Slide>
  ));

  return (
    <MainBox>
    {
      isLoading 
        ? <Spiner />
        : (<>
            <SlideshowBox ref={slideshow}>
              {banners}
            </SlideshowBox>
            <Controls>
              <Button onClick={lastPic}>
                <LeftArrowIcon />
              </Button>
              <Button right onClick={nextPic}>
                <RightArrowIcon />
              </Button>
            </Controls>
          </>)
    }
    </MainBox>
  );
}

export default Slider;
