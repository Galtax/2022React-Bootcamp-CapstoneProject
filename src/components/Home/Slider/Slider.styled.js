import styled from 'styled-components';

export const MainBox = styled.div`
  overflow: hidden;
  margin: 0 auto;
  position: relative;
 `;

export const SlideshowBox = styled.div`
  display: flex;
  flex-wrap: nowrap;
`;

export const Slide = styled.div`
  box-sizing: border-box;
  min-width: 100%;
  overflow: hidden;
  transition: .3s ease all;
  z-index: 10;
  max-height: 500px;
  position: relative;

  img {
    width: 100%;
    height: fit-content;
  }

  @media screen and (max-width: 1150px){
    max-height: 400px;
  }

  @media screen and (max-width: 926px){
    max-height: 300px;
  }

  @media screen and (max-width: 700px){
    max-height: 220px;
  }
`;

export const Controls = styled.div`
  position: absolute;
  top: 0;
  z-index: 20;
  width: 100%;
  height: 100%;
  pointer-events: none;
`;

export const Button = styled.button`
  pointer-events: all;
  background: none;
  border: none;
  cursor: pointer;
  outline: none;
  width: 50px;
  height: 100%;
  text-align: center;
  position: absolute;
  transition: all .3s ease;
  
  &:hover {
    background: hsla(189, 44%, 25%, 0.4);
    path {
      fill: #fff;
    }
  }

  path {
    filter: ${(props) => (props.right
    ? 'drop-shadow(-1px 0px 0px #fff)'
    : 'drop-shadow(1px 0px 0px #fff)')};
  }

  ${(props) => (props.right ? 'right: 0' : 'left: 0')}
`;
