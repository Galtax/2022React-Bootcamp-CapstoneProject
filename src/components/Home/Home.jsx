import React from 'react';
import { Link } from 'react-router-dom'
import { Button, ButtonWrapper } from '../../App.style';
import { ROUTES } from '../../utils/constants';
import FeaturedProducts from './FeaturedProducts/FeaturedProducts';
import { HomeMain } from './Home.styled';
import ProductCategories from './ProductCategories';
import Slider from './Slider';

function Home() {
  return(
    <HomeMain>
      <Slider />
      <ProductCategories />
      <FeaturedProducts />
      <ButtonWrapper>
        <Link to={ROUTES.productList} >
          <Button >
            View all Products
          </Button>
        </Link>
      </ButtonWrapper>
    </HomeMain>
  );
}

export default Home;