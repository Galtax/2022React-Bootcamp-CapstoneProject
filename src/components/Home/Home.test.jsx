import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Home from '.';
import Slider from './Slider';
import ProductCategories from './ProductCategories';
import FeaturedProducts from './FeaturedProducts/FeaturedProducts';

describe('Home Component Test', () => {

  it('should render the Home Component correctly', () => {
    const wrapper = mount(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should contain subcomponents', () => {
    const wrapper = mount(<Home />);
    
    expect(wrapper.find(Slider).exists()).toEqual(true);
    expect(wrapper.find(ProductCategories).exists()).toEqual(true);
    expect(wrapper.find(FeaturedProducts).exists()).toEqual(true);
  });
});
