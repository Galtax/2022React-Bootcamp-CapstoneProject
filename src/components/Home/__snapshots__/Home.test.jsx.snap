// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Test should render the Home Component correctly 1`] = `ReactWrapper {}`;
